nixos/modules/services/networking/globalprotect-vpn.nix

```diff
@@ -0,0 +1,43 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.globalprotect;
+
+  execStart = if cfg.csdWrapper == null then
+      "${pkgs.globalprotect-openconnect}/bin/gpservice"
+    else
+      "${pkgs.globalprotect-openconnect}/bin/gpservice --csd-wrapper=${cfg.csdWrapper}";
+in
+
+{
+  options.services.globalprotect = {
+    enable = mkEnableOption "globalprotect";
+
+    csdWrapper = mkOption {
+      description = ''
+        A script that will produce a Host Integrity Protection (HIP) report,
+        as described at http://www.infradead.org/openconnect/hip.html
+      '';
+      default = null;
+      example = "${pkgs.openconnect}/libexec/openconnect/hipreport.sh";
+      type = types.nullOr types.path;
+    };
+  };
+
+  config = {
+    services.dbus.packages = [ pkgs.globalprotect-openconnect ];
+
+    systemd.services.gpservice = {
+      description = "GlobalProtect openconnect DBus service";
+      serviceConfig = {
+        Environment="\"LANG=en_US.utf8\"";
+        Type="dbus";
+        BusName="com.yuezk.qt.GPService";
+        ExecStart=execStart;
+      };
+      wantedBy = [ "multi-user.target" ];
+    };
```

Comment:
How about something like this:
```
systemd.packages = [ pkgs.globalprotect-openconnect ];
systemd.services.gpservice = {
  wantedBy = [ "multi-user.target" ];
  serviceConfig.ExecStart = mkIf (cfg.csdWrapper != null) [ "" "${pkgs.globalprotect-openconnect}/bin/gpservice --csd-wrapper=${cfg.csdWrapper}" ];
};
```

Reply:
I passed on this suggestion, hope that's okay!  :)

Reply:
I'm confused... this is your PR. Who did you pass this along to?

Reply:
Sorry, I meant "passed on" as in "decided not to take".  The way I have it reads more clearly to me, so I exercised the dealer's choice and elected to leave it that way.

Reply:
Oh, ha! Yes, as mentioned dealers choice.

😄